src/Model/GroundWaterFlow/gwf3lak8.f90

```diff
@@ -4884,139 +4902,149 @@ subroutine lak_rp_obs(this)
+            nn1 =  obsrv%indxbnds(j)
+            if (nn1 < 1 .or. nn1 > this%noutlets) then
+              write(errmsg, '(a,1x,a,1x,i0,1x,a,1x,i0,a)')                       &
+                trim(adjustl(obsrv%ObsTypeId)),                                  &
+                ' outlet must be > 0 and <=', this%noutlets,                     &
+                '(specified value is ', nn1, ')'
+              call store_error(errmsg)
+            end if
+          end do
+        else if (obsrv%ObsTypeId=='LAK' .or.                                     &
+                 obsrv%ObsTypeId=='CONDUCTANCE' .or.                             &
+                 obsrv%ObsTypeId=='WETTED-AREA') then
+          !do j = 1, size(obsrv%indxbnds)
+          do j = 1, obsrv%indxbnds_count
+            nn1 =  obsrv%indxbnds(j)
+            if (nn1 < 1 .or. nn1 > this%maxbound) then
+              write(errmsg, '(a,1x,a,1x,i0,1x,a,1x,i0,a)')                       &
+                trim(adjustl(obsrv%ObsTypeId)),                                  &
+                'lake connection number must be > 0 and <=', this%maxbound,      &
+                '(specified value is ', nn1, ')'
+              call store_error(errmsg)
+            end if
+          end do
         else
-          errmsg = 'Programming error in lak_rp_obs'
-          call store_error(errmsg)
-        endif
-      end if
-      !
-      ! -- catch non-cumulative observation assigned to observation defined
-      !    by a boundname that is assigned to more than one element
-      if (obsrv%ObsTypeId == 'STAGE') then
-        n = size(obsrv%indxbnds)
-        if (n > 1) then
-          write(errmsg, '(a,3(1x,a))')                                           &
-            trim(adjustl(obsrv%ObsTypeId)),                                      &
-            'for observation', trim(adjustl(obsrv%Name)),                        &
-            ' must be assigned to a lake with a unique boundname.'
-          call store_error(errmsg)
+          !do j = 1, size(obsrv%indxbnds)
+          do j = 1, obsrv%indxbnds_count
+            nn1 =  obsrv%indxbnds(j)
+            if (nn1 < 1 .or. nn1 > this%nlakes) then
+              write(errmsg, '(a,1x,a,1x,i0,1x,a,1x,i0,a)')                       &
+                trim(adjustl(obsrv%ObsTypeId)),                                  &
+                ' lake must be > 0 and <=', this%nlakes,                         &
+                '(specified value is ', nn1, ')'
+              call store_error(errmsg)
+            end if
+          end do
         end if
-      end if
+      end do
       !
-      ! -- check that index values are valid
-      if (obsrv%ObsTypeId=='TO-MVR' .or.                                         &
-          obsrv%ObsTypeId=='EXT-OUTFLOW' .or.                                    &
-          obsrv%ObsTypeId=='OUTLET') then
-        do j = 1, size(obsrv%indxbnds)
-          nn1 =  obsrv%indxbnds(j)
-          if (nn1 < 1 .or. nn1 > this%noutlets) then
```

Comment:
We had a bug with this before, and I think you may have introduced it again here.  For some obs, we have to check against NOUTLETS, not MAXBOUND, or NLAKES.  I can't tell if that's been done here.

Reply:
The code is the same except for `size(obsrv%indxbnds)` is replaced by `obsrv%indxbnds_count`